sql/sql_connect.cc

```diff
@@ -346,6 +349,68 @@ void init_global_index_stats(void)
   }
 }
 
+void reset_global_user_or_client_stats(HASH* stats)
+{
+  DYNAMIC_ARRAY delete_items;
+
+  if (my_init_dynamic_array(&delete_items, sizeof(USER_STATS *),
+        stats->records, 1))
+  {
+    sql_print_error("reset global stats: cannot initialize dynamic array");
+    my_message(ER_OUTOFMEMORY, ER(ER_OUTOFMEMORY), MYF(0));
+    return;
+  }
+
+  for (uint idx=0;idx < stats->records; idx++)
+  {
+    USER_STATS* user_stats=
+      (USER_STATS *) my_hash_element(stats, idx);
+    if (user_stats->concurrent_connections)
+    {
+      init_user_stats(user_stats, NULL, NULL,
```

Comment:
This function has way too many zero arguments - this is not the part of this bugfix, but I have to ask you to clean it up anyway: please only keep the arguments that passed here or at the other call in increment_count_by_name, and move the rest of zero initialisation to the function body. Please also make it "static" and remove its prototype from sql/structs.h